Content.Server/Chemistry/Components/InjectorComponent.cs

```diff
@@ -163,6 +186,51 @@ async Task<bool> IAfterInteract.AfterInteract(AfterInteractEventArgs eventArgs)
             return true;
         }
 
+        /// <summary>
+        /// Send informative pop-up messages and wait for a Do-After to complete.
+        /// </summary>
+        public async Task<bool> TryInjectDoAfter(EntityUid user, EntityUid target)
+        {
+            InUse = true;
+            var popupSys = EntitySystem.Get<SharedPopupSystem>();
+
+            // for the user
+            popupSys.PopupEntity(Loc.GetString("injector-component-injecting-user"), target, Filter.Entities(user));
+
+            if (user != target)
+            {
+                // for the target
+                var userName = Owner.EntityManager.GetComponent<MetaDataComponent>(user).EntityName;
+                popupSys.PopupEntity(Loc.GetString("injector-component-injecting-target",
+                    ("user", userName)), user, Filter.Entities(target));
+
+                // also, lets log this
+                var userEntity = Owner.EntityManager.GetEntity(user);
+                var targetEntity = Owner.EntityManager.GetEntity(target);
+
+                var text = ToggleState == InjectorToggleMode.Inject
+                    ? "inject reagents into"
+                    : "draw reagents from";
+
+                // using the "force feed" log type. It's not quite feeding, but the effect is the same.
+                EntitySystem.Get<AdminLogSystem>().Add(LogType.ForceFeed,
+                    $"{userEntity} is attempting to {text} {targetEntity}");
```

Comment:
Would it be more useful to just log all injections so that suicides can be checked too? Or would that be too much spam with regular healing?

Reply:
I guess logging all injections makes sense, I'll make it do that. But If you wanted to log all chem intake, you'd also need to start logging all eating/drinking (currently its just logging force-feeding/drinking). In order to avoid spamming logs with generic nutriment & water intake anytime anyone eats anything, I'll probably make it log only non-nutriemtn/water intake (+ force feeding).

#5569 is gonna add a solution `ToPrettyString()` function for logging, and I was going to go back through all of these logs anyway once that is merged, so that the logs actually show reagent info.

Reply:
I've added the a `LogType.Ingestion` for self-injections, which I'll eventually add to food & drink.

I'm also no longer logging it when people draw a solution from somebody, because that doesn't really belong in the force-feed or ingestion categories.  Currently that can't really do any harm anyways? I mean, I guess people could try steal your vodka from your bloodstream if they can't afford their own, but probably doesn't need to be logged yet.